types/node/https.d.ts

```diff
@@ -25,14 +26,56 @@ declare module 'https' {
         constructor(options?: AgentOptions);
         options: AgentOptions;
     }
-    interface Server extends http.HttpBase {}
+    interface Server extends http.Server {}
     /**
      * See `http.Server` for more information.
      * @since v0.3.4
      */
     class Server extends tls.Server {
         constructor(requestListener?: http.RequestListener);
         constructor(options: ServerOptions, requestListener?: http.RequestListener);
+        addListener(event: string, listener: (...args: any[]) => void): this;
```

Comment:
Note: Current TS limitation is that inherited events from tls.Server (and net.Server respectively) will be overriden when we create the overloads here.

I don't expect you to do fix it since it would require copy/pasting all the overloads but just noting.

I plan to eventually setup an auto generator for these events that handles the copy pasting :)

Reply:
- Add `net` server events to `http` [d355807](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54886/commits/d355807973418aa2d1cfe0a2786ae52c1b543e57)
- Add `net` server events to `https` [3c4eb3c](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54886/commits/3c4eb3c2eeb74877070adebf06e750154954d302)
- Add `tls` server events to `https` [7abda3d](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54886/commits/7abda3dcdebcfa21606dff631edb3dfd0774db5f)